From 170fa86b722e1ffcea1068ca357bc8b12da9ba75 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Wed, 11 Jan 2017 13:43:53 -0600
Subject: [PATCH 663/895] soc: ti: wkup_m3_ipc: Clean up pm_ops to prevent
 build warnings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Wrap wkup_m3_ipc_resume in CONFIG_PM_SLEEP instead of CONFIG_PM and move
the pm_ops struct outside of the ifdef to avoid the ifdef later in the
platform_driver definition. Otherwise we see this warning on build
with !CONFIG_SUSPEND:

drivers/soc/ti/wkup_m3_ipc.c:648:12: warning: ‘wkup_m3_ipc_resume’
defined but not used [-Wunused-function]
 static int wkup_m3_ipc_resume(struct device *dev)

Fixes: 85b7e7b2abb6 ("soc: ti: wkup_m3_ipc: Add rtc_only support")
Reported-by: Suman Anna <s-anna@ti.com>
Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
(cherry picked from commit fc2eb0a717db5a1e6ce1659db3783ba1f09fd88a)
---
 drivers/soc/ti/wkup_m3_ipc.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/soc/ti/wkup_m3_ipc.c b/drivers/soc/ti/wkup_m3_ipc.c
index edfe5a8..ed44950 100644
--- a/drivers/soc/ti/wkup_m3_ipc.c
+++ b/drivers/soc/ti/wkup_m3_ipc.c
@@ -722,7 +722,7 @@ static int wkup_m3_ipc_remove(struct platform_device *pdev)
 	return 0;
 }
 
-#ifdef CONFIG_PM
+#ifdef CONFIG_PM_SLEEP
 static int wkup_m3_ipc_resume(struct device *dev)
 {
 	if (m3_ipc_state->is_rtc_only) {
@@ -734,11 +734,11 @@ static int wkup_m3_ipc_resume(struct device *dev)
 
 	return 0;
 }
+#endif /* CONFIG_PM_SLEEP */
 
 static const struct dev_pm_ops wkup_m3_ipc_pm_ops = {
 	SET_SYSTEM_SLEEP_PM_OPS(NULL, wkup_m3_ipc_resume)
 };
-#endif
 
 static const struct of_device_id wkup_m3_ipc_of_match[] = {
 	{ .compatible = "ti,am3352-wkup-m3-ipc", },
@@ -753,9 +753,7 @@ static struct platform_driver wkup_m3_ipc_driver = {
 	.driver = {
 		.name = "wkup_m3_ipc",
 		.of_match_table = wkup_m3_ipc_of_match,
-#ifdef CONFIG_PM
 		.pm = &wkup_m3_ipc_pm_ops,
-#endif
 	},
 };
 
-- 
1.7.5.4

